controllers/authControllers.js

```diff
@@ -0,0 +1,120 @@
+const { promisify } = require('util')
+const jwt = require('jsonwebtoken')
+const User = require('../models/userModel')
+
+const signToken = id => {
+    return jwt.sign(
+        { id },
+        process.env.JWT_SECRET,
+        {
+            expiresIn: process.env.JWT_EXPIRES_IN
+        }
+    )
+}
+
+exports.signup = async (req, res) => {
+    try {
+        const newUser = await User.create({
+            name: req.body.name,
+            email: req.body.email,
+            password: req.body.password,
+            passwordConfirm: req.body.passwordConfirm
+        })
+
+        const token = signToken(newUser._id)
+
+        res.status(201).json({
+            status: 'success',
+            token,
+            data: { newUser }
+        })
+    } catch (error) {
+        res.status(404).json({
+            status: 'fail',
+            message: error
```

Comment:
You shouldn't expose the error to users, inspect the error object, and return a simplified but descriptive version.